model_analyzer/result/constraints_holder.py

```diff
@@ -0,0 +1,100 @@
+# Copyright (c) 2021-2022, NVIDIA CORPORATION. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+GLOBAL_CONSTRAINTS_KEY = "default"
+
+class ConstraintsHolder:
+    """
+    This class holds model and global constraints
+    """
+    def __init__(self, config=None):
+        """
+        Parameters
+        ----------
+        config: ConfigCommandProfile/ConfigCommandReport or None
+            The model analyzer's config
+        """
+        self._model_constraints = {}
+        self._global_constraints = {}
+
+        if config:
+            if "profile_models" in config.get_config():
+                for model in config.profile_models:
+                    self._model_constraints[model.model_name()
+                                            ] = model.constraints()
+    
+            if "constraints" in config.get_all_config():
+                self._global_constraints[GLOBAL_CONSTRAINTS_KEY] = config.get_all_config()[
+                    "constraints"]
+
+    def empty(self):
+        """
+        Returns
+        -------
+        bool:
+            Returns True if no constraints are present, else returns False
+        """
+        if any(self._model_constraints.values()):
+            return False
+        else:
+            return True
+
+    def get_constraints_for_single_model(self, model_name):
+        """
+        Parameters
+        ----------
+        model_name: str
+            The name of the model
+
+        Returns
+        -------
+        model_constraints: None or dict
```

Comment:
I think we are missing a class here. We have a class to hold ALL constraints, but there should also be a class to hold a single model's constraints. Otherwise you have to pass all the constraints as well as the model name everywhere so that you can keep finding the model's constraints.

Also, we can avoid the "None or" part by always having a model constraint object for each model, but allowing it to have no constraints in it.